### ISSUE TITLE
Credentials

### ISSUE THREAD

**Issue by 07kamil** [microsoft/azuredatastudio#22541]
Issue Type: <b>Bug</b>

Hi
I cannot enter my credentials intp app, hance i cannot use and query DB
When i try to connect to DB wheel is spinning but never ends
Best
Kamil

Azure Data Studio version: azuredatastudio 1.42.0 (22f384e569bc90d8b3231cc8c8cff99ea16c5052, 2023-03-17T18:15:48.982Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No
Preview Features: Enabled

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz (16 x 2496)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|63.73GB (36.56GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
admin-pack|Mic|0.0.2
agent|Mic|0.49.0
arc|Mic|1.8.0
azcli|Mic|1.8.0
cms|Mic|0.9.2
dacpac|Mic|1.13.0
datavirtualization|Mic|1.13.1
import|Mic|1.5.4
profiler|Mic|0.12.2

</details>

**Comment by Duenke** [comment]
Adding to this, the recent update seems to have involved the new `accessTokenCache` change.
As for me, I had to go into settings and ensure settings used `MSAL`, and `Code Grant Method` (both I already had), and then tick the setting `mssql:enableSqlAuthenticationProvider`.

Only then could I successfully authenticate to Azure SQL.
The problem now, though, is that the `accessTokenCache` file is written to my open git repository. Not ideal for keeping my tokens to myself.
I did scan through recent code changes, and it's not immediately obvious to me **why** it does this. The code seems to point to writing the file to AppData.

**Comment by cheenamalhotra** [comment]
Hi @07kamil 

Please verify Azure Core extension is enabled as it might be disabled.. we've heard reports for the same issue. 
https://github.com/microsoft/azuredatastudio/wiki/Troubleshoot-Azure-Authentication-Issues#ensure-azure-core-extension-is-enabled

**Comment by cheenamalhotra** [comment]
Hi @Duenke 

Can you provide platform details and which installation of ADS are you working with?

**Comment by Duenke** [comment]
Hey @cheenamalhotra 

Version: 1.42.0 (user setup)
Commit: 22f384e569bc90d8b3231cc8c8cff99ea16c5052
Date: 2023-03-17T18:15:48.982Z
VS Code: 1.67.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621

I'll also add that the `accessTokenCache` files is also being written and updated on the `AppData\Roaming\azuredatastudio\Azure Accounts` directory as expected.
I also tried opening ADS with no folder/git repo open and it didn't seem to write the duplicate file anywhere.
I also tried opening ADS with some random non-git folder and it did again write the duplicate file.

The triggers are the same time the expected file is written. Any attempt to connect to Azure SQL.

Azure core is installed and enabled, and here are the auth settings I mentioned above:
<img width="564" alt="image" src="https://user-images.githubusercontent.com/46510548/228935956-b1e68cd5-c5ca-45f6-a691-40871b5b47d9.png">
